shared/middleware/helpers/storage_apps.rb

```diff
@@ -140,24 +140,44 @@ def get_abuse(channel_id)
 
   # Determine if the current user can view the project
   def get_sharing_disabled(channel_id, current_user_id)
-    user_storage_owner_id, _ = storage_decrypt_channel_id(channel_id)
-    owner_id = user_storage_ids_table.where(id: user_storage_owner_id).first[:user_id]
+    owner_storage_id, storage_app_id = storage_decrypt_channel_id(channel_id)
+    owner_user_id = user_storage_ids_table.where(id: owner_storage_id).first[:user_id]
 
     # Sharing of a project is not disabled for the project owner
```

Comment:
quick optimization: at this point it seems like you could `return false unless current_user_id`

Reply:
oops I'm wrong, sharing might be enabled or disabled based on `get_user_sharing_disabled`

Reply:
agreed, that shouldn't be needed.